### ISSUE TITLE
trouble impl'ing multiple typetag traits

### ISSUE THREAD

**Issue by ijackson** [dtolnay/typetag#28]
Hi.  I love this crate and am using it lots.  I have stumbled across a limitation:
```
pub use serde::{Serialize,Deserialize};

#[typetag::serde]
pub trait Base : Send { fn base(&self) -> usize; }

#[derive(Serialize,Deserialize)]
struct SomeBase;

#[typetag::serde]
impl Base for SomeBase { fn base(&self) -> usize { 12 } }


#[typetag::serde]
pub trait Top : Send + Base { }

#[derive(Serialize,Deserialize)]
struct WrappingTop(Box<dyn Base>);

#[typetag::serde]
impl Base for WrappingTop { fn base(&self) -> usize { self.0.base() } }

#[typetag::serde]
impl Top for WrappingTop { }
```
This produces:
```
13 | #[typetag::serde]
   | ^^^^^^^^^^^^^^^^^ multiple `typetag_name` found
...
help: disambiguate the associated function for candidate #1
   |
13 | Base::typetag_name
   |
help: disambiguate the associated function for candidate #2
   |
13 | Top::typetag_name
 ```
I think what I am asking typetag and serde to do here makes sense.  Making it possible might even be as simple as disambiguating the names in the macro expansion...

In the meantime I have a workaround which looks roughly like this:
```
pub use serde::{Serialize,Deserialize};

pub trait Base : Send { fn base(&self) -> usize; }

#[typetag::serde]
pub trait SerdeBase : Base { }

#[derive(Serialize,Deserialize)]
struct SomeBase;

impl Base for SomeBase { fn base(&self) -> usize { 12 } }

#[typetag::serde]
impl SerdeBase for SomeBase { }


#[typetag::serde]
pub trait Top : Send + Base { }

#[derive(Serialize,Deserialize)]
struct WrappingTop(Box<dyn SerdeBase>);

impl Base for WrappingTop { fn base(&self) -> usize { self.0.base() } }

#[typetag::serde]
impl SerdeBase for WrappingTop { }

#[typetag::serde]
impl Top for WrappingTop { }
```

**Comment by dtolnay** [comment]
Fixed in 0.1.6.

**Comment by ijackson** [comment]
David Tolnay writes ("Re: [dtolnay/typetag] trouble impl'ing multiple typetag traits (#28)"):
> Fixed in 0.1.6.

Thanks for the quick fix :-).

Ian.

-- 
Ian Jackson <ijackson@chiark.greenend.org.uk>   These opinions are my own.  

Pronouns: they/he.  If I emailed you from @fyvzl.net or @evade.org.uk,
that is a private address which bypasses my fierce spamfilter.
